packages/components/src/components/hds/advanced-table/index.ts

```diff
@@ -0,0 +1,346 @@
+      return `Sorted by ${this._sortBy} ${this._sortOrder}ending`;
+    } else {
+      return '';
+    }
+  }
+
+  get isSelectable(): boolean {
+    const { isSelectable = false } = this.args;
+
+    return isSelectable;
+  }
+
+  get isStriped(): boolean {
+    const { isStriped = false } = this.args;
+
+    return isStriped;
+  }
+
+  get density(): HdsAdvancedTableDensities {
+    const { density = DEFAULT_DENSITY } = this.args;
+
+    assert(
+      `@density for "Hds::Table" must be one of the following: ${DENSITIES.join(
+        ', '
+      )}; received: ${density}`,
+      DENSITIES.includes(density)
+    );
+
+    return density;
+  }
+
+  get valign(): HdsAdvancedTableVerticalAlignment {
+    const { valign = DEFAULT_VALIGN } = this.args;
+
+    assert(
+      `@valign for "Hds::Table" must be one of the following: ${VALIGNMENTS.join(
+        ', '
+      )}; received: ${valign}`,
+      VALIGNMENTS.includes(valign)
+    );
+
+    return valign;
+  }
+
+  get gridTemplateColumns(): string {
+    if (!this.args.isSelectable && !this.columnWidths) {
+      return `repeat(${this.args.columns.length}, 1fr)`;
+    }
+
+    let style = this.args.isSelectable ? 'auto' : '';
+
+    if (this.columnWidths) {
+      for (let i = 0; i < this.columnWidths.length; i++) {
+        style = `${style} ${this.columnWidths[i] ? this.columnWidths[i] : '1fr'}`;
+      }
+    } else {
+      for (let i = 0; i < this.args.columns.length; i++) {
+        style = `${style} 1fr`;
+      }
+    }
+
+    return style;
+  }
+
+  get classNames(): string {
+    const classes = ['hds-advanced-table'];
+
+    // add a class based on the @isStriped argument
+    if (this.isStriped) {
+      classes.push('hds-advanced-table--striped');
+    }
+
+    // add a class based on the @density argument
+    if (this.density) {
+      classes.push(`hds-advanced-table--density-${this.density}`);
+    }
+
+    // add a class based on the @valign argument
+    if (this.valign) {
+      classes.push(`hds-advanced-table--valign-${this.valign}`);
+    }
+
+    return classes.join(' ');
+  }
+
+  @action
+  setSortBy(column: string): void {
+    if (this._sortBy === column) {
+      // check to see if the column is already sorted and invert the sort order if so
+      this._sortOrder =
+        this._sortOrder === HdsAdvancedTableThSortOrderValues.Asc
+          ? HdsAdvancedTableThSortOrderValues.Desc
+          : HdsAdvancedTableThSortOrderValues.Asc;
+    } else {
+      // otherwise, set the sort order to ascending
+      this._sortBy = column;
+      this._sortOrder = HdsAdvancedTableThSortOrderValues.Asc;
+    }
+
+    const { onSort } = this.args;
+
+    if (typeof onSort === 'function') {
+      onSort(this._sortBy, this._sortOrder);
+    }
+  }
+
+  onSelectionChangeCallback(
+    checkbox?: HdsFormCheckboxBaseSignature['Element'],
+    selectionKey?: string
+  ): void {
+    const { onSelectionChange } = this.args;
+    if (typeof onSelectionChange === 'function') {
```

Comment:
[suggestion][nitpick] - what do you think about doing the inverse check here and returning early? imo less nesting is easier to read.